Code/EnginePlugins/PhysXPlugin/WorldModule/Implementation/PhysXWorldModule.cpp

```diff
@@ -655,11 +648,41 @@ ezUInt32 ezPhysXWorldModule::CreateShapeId()
   return m_uiNextShapeId++;
 }
 
-void ezPhysXWorldModule::DeleteShapeId(ezUInt32 uiShapeId)
+void ezPhysXWorldModule::DeleteShapeId(ezUInt32& uiShapeId)
 {
-  EZ_ASSERT_DEV(uiShapeId != ezInvalidIndex, "Trying to delete an invalid shape id");
+  if (uiShapeId == ezInvalidIndex)
+    return;
 
   m_FreeShapeIds.PushBack(uiShapeId);
+
+  uiShapeId = ezInvalidIndex;
+}
+
+ezUInt32 ezPhysXWorldModule::AllocateUserData(ezPxUserData*& out_pUserData)
```

Comment:
Are these functions thread safe?

Reply:
No they aren't, but they don't need to be since the whole component init/deinit is synchronous anyways.